enhancements/windows-containers/bring-your-own-windows-host.md

```diff
@@ -0,0 +1,196 @@
+---
+title: bring-your-own-windows-host
+authors:
+  - "@aravindhp"
+reviewers:
+  - "@@openshift/openshift-team-windows-containers"
+approvers:
+  - "@sdodson"
+creation-date: 2021-01-19
+last-updated: 2021-01-19
+status: implementable
+---
+
+# Bring Your Own Windows Host
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow customers to add their existing
+Windows instances to an OpenShift 4.7+ cluster using the Windows Machine Config
+Operator (WMCO).
+
+## Motivation
+
+The world of Windows sees customers treating their instances more as "pets"
+rather than "cattle". There is a desire from customers to be able to reuse these
+"pet" Windows instances as OpenShift worker nodes, run Windows workloads and gain
+similar benefits that their Linux workloads get when being managed by OpenShift.
+
+### Goals
+
+* Allow a customer to add an existing Windows instance running a variant of
+  Windows Server 2019 (1809, 1909, 20H2 etc). The instance should have the
+  Docker runtime and SSH services configured. The public key used by WMCO for
+  the [IPI workflow](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator.md#design-details)
+  needs to be an authorized key in the instance.
+* WMCO will then perform all the required steps within the VM for it to be
+  added to the cluster as an OpenShift worker node and removed from it.
+* [Handle upgrades](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+  of all components installed by WMCO.
+
+### Non-Goals
+
+* Installing the container runtime in the Windows instance
+* Managing the Windows operating system(OS) including handling of OS updates
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The cluster admin will create a ConfigMap called _windows-instances_ in the
+_openshift-windows-machine-config-operator_ namespace. WMCO will be configured
+to watch for this special ConfigMap, whose data section will have the following
+layout:
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: windows-instances
+  namespace: openshift-windows-machine-config-operator
+data:
+  10.1.42.1: |-
+    username=Administrator
+  instance.dns.com: |-
```

Comment:
IPs or DNS will be used as a connection string or ID. If the customer specifies an IP entry and a DNS entry for the same instance then it will get prepped twice but will result in only one worker. I will call out that this entry needs to be unique.